client/app/services/query.js

```diff
@@ -103,6 +136,39 @@ class Parameter {
   set ngModel(value) {
     this.setValue(value);
   }
+
+  toQueryParams() {
+    if (this.isEmpty) {
+      return {};
+    }
+    if (isDateRangeParameter(this.type)) {
+      return {
+        [`p_${this.name}.start`]: this.value.start,
+        [`p_${this.name}.end`]: this.value.end,
+      };
+    }
+    return {
+      [`p_${this.name}`]: this.value,
+    };
+  }
+
+  fromQueryParams(query) {
+    if (isDateRangeParameter(this.type)) {
+      this.setValue([
+        query[`p_${this.name}.start`],
+        query[`p_${this.name}.end`],
+      ]);
+    } else {
+      this.setValue(query[`p_${this.name}`]);
+    }
+  }
+
+  toQuerySnippet() {
```

Comment:
We already have a concept named Query Snippet in the code base, so it might be confusing to have the same name here. Maybe `toQueryText`?